Services/LegalDocuments/classes/Setup/Agent.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace ILIAS\LegalDocuments\Setup;
+
+use ILIAS\Refinery\Factory as Refinery;
+use ILIAS\Refinery\Transformation;
+use ILIAS\Setup\Objective;
+use ILIAS\Setup\ObjectiveCollection;
+use ILIAS\Setup\Config;
+use ILIAS\Setup\Metrics\Storage;
+use ILIAS\Setup\Objective\NullObjective;
+use ILIAS\Setup\Agent as AgentInterface;
+use ilDatabaseUpdateStepsExecutedObjective;
+
+class Agent implements AgentInterface
+{
+    public function __construct(private readonly Refinery $refinery)
+    {
+    }
+
+    public function hasConfig(): bool
+    {
+        return false;
+    }
+
+    public function getArrayToConfigTransformation(): Transformation
+    {
+        return $this->refinery->identity();
+    }
+
+    public function getInstallObjective(Config $config = null): Objective
+    {
+        return new NullObjective();
+    }
+
+    public function getUpdateObjective(Config $config = null): Objective
+    {
+        return $this->list([
+            new ilDatabaseUpdateStepsExecutedObjective(new UpdateSteps()),
+            new ConsumerObjective(),
```

Comment:
A minor addition: you should return the `ConsumerObjective` in `getBuildArtifactObjective()` instead and leave it up to this service to decide at what point the artifact should be built. Otherwise this artifact will also be ignored when the `build-artifacts` command is run.

Reply:
@thibsy: Done.